tests/unit/plugins/service/test_gdns_publisher.py

```diff
@@ -308,56 +275,43 @@ def test_implements_interface(mocker, config, metrics):
 @pytest.mark.asyncio
 async def test_handle_message_handles_update_conflict(
         gdns_publisher_instance, event_message,
-        matching_zone_records, rrsets_url, changes_url,
         initial_changes_req, handled_conflict_changes_req,
         initial_changes_pending_json_resp, initial_changes_resp, caplog,
-        mocker):
+        mocker, mock_dns_client):
     """Ensure changes with update conflicts are successfully published"""
-    expected_change_id = json.loads(initial_changes_pending_json_resp)['id']
+    # expected_change_id = json.loads(initial_changes_pending_json_resp)['id']
     event_message.data['resourceRecords'] = initial_changes_req['additions']
-
-    gdns_publisher_instance.http_client._request_post_mock.side_effect = [
+    mock_dns_client._publish_changes_mock.side_effect = [
         exceptions.GCPHTTPResponseError('409', 409),
         initial_changes_pending_json_resp]
-    gdns_publisher_instance.http_client._get_json_mock.side_effect = [
-        matching_zone_records, initial_changes_resp]
 
     await gdns_publisher_instance.handle_message(event_message)
-
-    expected_get_json_calls = [
-        mocker.call(rrsets_url),
-        mocker.call(f'{changes_url}/{expected_change_id}')
-    ]
-
-    gdns_publisher_instance.http_client._get_json_mock.has_calls(
-        expected_get_json_calls)
-    gdns_publisher_instance.http_client._request_post_mock.assert_called_with(
-        'post', changes_url, json=handled_conflict_changes_req)
+    mock_dns_client._publish_changes_mock.assert_called_with(
+        'example.com.', handled_conflict_changes_req)
     assert 3 == len(caplog.records)
 
 
 @pytest.mark.asyncio
 async def test_handle_message_raises_exception_on_publish_timeout(
-        gdns_publisher_instance, event_message, changes_url,
-        initial_changes_req, initial_changes_resp,
-        initial_changes_pending_json_resp, mock_sleep, caplog):
+        gdns_publisher_instance, event_message, mock_dns_client,
+        initial_changes_req, mock_sleep, caplog, mocker):
     """Ensure exception raised when publish wait timeout exceeded."""
-    gdns_publisher_instance.publish_wait_timeout = 0.0001
+    start = datetime.datetime(2018, 1, 1, 11, 30, 0)
+    mockdatetime = mocker.Mock()
+    mockdatetime.now = mocker.Mock(side_effect=[
+        start, start, start + datetime.timedelta(seconds=100)])
```

Comment:
Would be nice to replace 100 with gdns_publisher_instance.config['publish_wait_timeout'] + 1.